truffle/src/main/java/org/jruby/truffle/core/string/StringLiteral.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (c) 2016 Oracle and/or its affiliates. All rights reserved. This
+ * code is released under a tri EPL/GPL/LGPL license. You can use it,
+ * redistribute it and/or modify it under the terms of the:
+ *
+ * Eclipse Public License version 1.0
+ * GNU General Public License version 2
+ * GNU Lesser General Public License version 2.1
+ */
+package org.jruby.truffle.core.string;
+
+import com.oracle.truffle.api.CompilerDirectives;
+import com.oracle.truffle.api.CompilerDirectives.CompilationFinal;
+import com.oracle.truffle.api.object.DynamicObject;
+import org.jcodings.specific.USASCIIEncoding;
+import org.jruby.truffle.RubyContext;
+import org.jruby.truffle.core.rope.CodeRange;
+import org.jruby.truffle.core.rope.Rope;
+
+import java.nio.charset.StandardCharsets;
+
+public class StringLiteral {
+
+    private final RubyContext context;
+    private final String literal;
+
+    @CompilationFinal private volatile Rope rope;
+
+    public StringLiteral(RubyContext context, String literal) {
+        assert is7Bit(literal);
+        this.context = context;
+        this.literal = literal;
+    }
+
+    public Rope getRope() {
+        if (rope == null) {
+            CompilerDirectives.transferToInterpreterAndInvalidate();
+            rope = createRope();
+        }
+
+        return rope;
+    }
+
+    public DynamicObject getString() {
+        return StringOperations.createString(context, getRope());
+    }
+
+    private Rope createRope() {
+        // getRope is fully synchronised and returns identically the same object for the same parameters
+
+        return context.getRopeTable().getRope(
+                literal.getBytes(StandardCharsets.US_ASCII),
+                USASCIIEncoding.INSTANCE,
```

Comment:
This would be for correctness, however. In the `defined?` case, for instance, MRI would return a UTF-8 string. We can have a 7-bit UTF-8 rope (that's the dominant case in normal operation).